geospaas/base_viewer/tests.py

```diff
@@ -0,0 +1,127 @@
+from html.parser import HTMLParser
+
+from django.test import Client, TestCase
+from django.utils import timezone
+from mock.mock import MagicMock, patch
+
+from geospaas.base_viewer.views import IndexView
+
+
+class MyHTMLParser(HTMLParser):
+    def __init__(self):
+        """Constructor with extra attribute definition"""
+        super().__init__()
+        self.flag = False
+        self.data = []
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'td':  # find out the tags with td tags
+            for attr in attrs:
+                # find out the ones with ///class="place_ds"///
+                if ('class' in attr) and ('place_ds' in attr):
+                    self.flag = True # make the flag true to enable the storage action
+
+    def handle_data(self, Data):
+        if (self.flag == True):# just store the data of specified tags with the help of flag
+            self.data.append(Data)
+            self.flag = False # make it false AGAIN in order not to save the data from other tags of HTML
+
+
+class IntegrationTestsForGUIWithNewBase(TestCase):
+    '''INtegration tests for GET and POST methods of GUI'''
+    fixtures = ["vocabularies", "catalog"]
+
+    def setUp(self):
+        self.client = Client()
+        # this parser is configured to store the desired data of td tag into self.data
+        self.parser = MyHTMLParser()
+
+    def test_the_post_verb_of_GUI_with_proper_polygon(self):
+        """shall return only the first dataset of fixtures in the specified placement of datasets inside the resulted HTML
+        in the case of a POST request with a good choice of polygon"""
+        res1 = self.client.post('/', {'polygon': '{"type":"Polygon","coordinates":[[[0,0],[0,5],[5,5],[5,0],[0,0]]]}',
+                                      'time_coverage_start': timezone.datetime(2000, 12, 29),
+                                      'time_coverage_end': timezone.datetime(2020, 1, 1),
```

Comment:
It is not a very necessary consideration in the tests. Time lag between different places does not matter for tests.

Reply:
First of all, it seems pretty important that we test that datasets are correctly filtered based on dates.
Second, if you mix aware and naive datetimes, you will end up getting errors because you can't compare them.

Reply:
OK. Then let me see how precisely the previous tests (which are not written by me) are accomplishing regarding the time in the previous version of viewer and also check that if tzinfo matters for them or not. If you say it is important, then it is important.  :)